src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -541,4 +552,56 @@ public function assertTextInTableRowInRegion($text, $rowText, $region): void {
     }
   }
 
+  /**
+   * Take a screenshot.
+   *
+   * @Then I take screenshot
+   */
+  public function iTakeScreenshot(): void {
+    $this->saveScreenshot($this->getOutputFilename(), $this->getScreenshotParameter('path'));
```

Comment:
@kumikoono This can be changed to:

```suggestion
      $filename = $this->getOutputFilename()
        ?: sprintf('%s_%s_%s', $this->getMinkParameter('browser_name'), date('c'), uniqid('', true));

    $filepath = $this->getScreenshotParameter('path')
      ? $this->getScreenshotParameter('path')
      : (ini_get('upload_tmp_dir')
        ? ini_get('upload_tmp_dir')  // Please avoid the double ternary operators.
        : sys_get_temp_dir());

    try {
      if (is_dir($filepath)
        || mkdir($filepath, 0777, true)
        || is_dir($filepath)) {
        $filename = "${filename}.png";
      }
      echo "Screenshot at: "${filepath}/${filename}";
      $this->saveScreenshot($filename, filepath);
    } catch (\Behat\Mink\Exception\UnsupportedDriverActionException
      |\Behat\Mink\Exception\DriverException $e) {

      $data = $this->getSession()->getDriver()->getContent();
      $output_html = "${filepath}/${filename}.hml"
      echo "Screenshot at: ${output_html}";
      file_put_contents($output_html, $data);
    }
```

If saveScreenshot throws an exception, it still display a message "Screenshot at: "${filepath}/${filename}", but I think it should be fine since it means that taking a screenshot fails and also it shows an HTML.